src/datasets/arrow_dataset.py

```diff
@@ -331,66 +345,110 @@ def to_tf_dataset(
             label_cols (:obj:`List[str]` or :obj:`str`, default ``None``): Dataset column(s) to load as
                 labels. Note that many models compute loss internally rather than letting Keras do it, in which case it is
                 not necessary to actually pass the labels here, as long as they're in the input `columns`.
-            dummy_labels (:obj:`bool`, default ``False``): If no `label_cols` are set, output an array of "dummy" labels
-                with each batch. This can avoid problems with `fit()` or `train_on_batch()` that expect labels to be
-                a Tensor or np.ndarray, but should (hopefully) not be necessary with our standard train_step().
+            drop_columns (:obj:`List[str]` or :obj:`str` or :obj:`bool`, optional): Dataset column(s) to drop before
+                passing to the `collate_fn`. By default, no columns are dropped.
             prefetch (:obj:`bool`, default ``True``): Whether to run the dataloader in a separate thread and maintain
                 a small buffer of batches for training. Improves performance by allowing data to be loaded in the
                 background while the model is training.
+            error_on_missing (:obj:`bool`, default ``True``): Whether to return an error if one of the requested columns
+                is missing. Should be set to `False` if columns or label_cols are autodetected from a model.
```

Comment:
I'm leaving this bit, even though we're not passing the model anymore, because it's advice on what to do if the autodetection happened in another function!